tests/test_sessions.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+
+#   Copyright (c) 2010-2014, MIT Probabilistic Computing Project
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import pytest
+import bayeslite
+import bayeslite.sessions as ss
+import test_core
+import json
+import sqlite3
+
+def make_bdb():
+    crosscat = test_core.local_crosscat()
+    metamodel = test_core.CrosscatMetamodel(crosscat)
+    bdb = bayeslite.bayesdb_open(builtin_metamodels=False)
+    bayeslite.bayesdb_register_metamodel(bdb, metamodel)
+    return bdb
+
+def make_bdb_with_sessions():
+    bdb = make_bdb()
+    t = ss.SessionTracer(bdb)
+    return (bdb, t)
+
+def get_id(entry):
+    return entry[0]
+
+def get_session(entry):
+    return entry[1]
+
+def get_time(entry):
+    return entry[2]
+
+def get_type(entry):
+    return entry[3]
+
+def get_data(entry):
+    return entry[4]
+
+def get_num_sessions(executor):
+    return int(executor('''
```

Comment:
I feel like "executor(query).next()[0]" should get a name, perhaps in bayeslite, and we should all start using that and stop using this pattern. Thoughts?